docs/source/en/quick-start.md

```diff
@@ -51,45 +51,67 @@ full-length hash instead of the shorter 7-character commit hash:
 
 ```py
 >>> from huggingface_hub import login
 >>> login()
 ```
 
-It is also possible to login programmatically without being prompted to enter your token by directly
-passing the token to [`login`] like `login(token="hf_xxx")`. If you do so, be careful when
-sharing your source code. It is a best practice to load the token from a secure vault instead
-of saving it explicitly in your codebase/notebook.
+You can only be logged into one account at a time. Logging into a new account will automatically log you out of the previous one. To determine your currently active account, simply run the huggingface-cli whoami command.
+
+<Tip warning={true}>
+
+Once you are logged in, all requests to the Hub -even methods that don't necessarily require authentication- will use your access token by default. If you want to disable implicit use of your token, you should set `HF_HUB_DISABLE_IMPLICIT_TOKEN=1` as environment variable (see [reference](../package_reference/environment_variables#hfhubdisableimplicittoken)).
+
+</Tip>
+
+### Environment variable
+
+Another possibility to authenticate is to set the environment variable `HF_TOKEN` with your access token. This is especially useful in Spaces where you can set `HF_TOKEN` as a [Space secret](https://huggingface.co/docs/hub/spaces-overview#managing-secrets).
+
+<Tip>
+
+**NEW:** Google Colaboratory let you [define private keys](https://twitter.com/GoogleColab/status/1719798406195867814) for your notebooks. Define a `HF_TOKEN` secret to be automatically authenticated!
+
+</Tip>
+
+Authentication via an environment variable or a secret has priority over the token stored on your machine.
+
+### Programmatically
 
-You can be logged in only to 1 account at a time. If you login your machine to a new account, you will get logged out
-from the previous. Make sure to always which account you are using with the command `huggingface-cli whoami`.
-If you want to handle several accounts in the same script, you can provide your token when calling each method. This
-is also useful if you don't want to store any token on your machine.
+Finally, it is also possible to authenticate programmatically by passing your token to any method that accepts `token` as parameter. This is usually discouraged except in an environment where you don't want to store your token permanently or if you need to handle several tokens at once.
 
 <Tip warning={true}>
 
-Once you are logged in, all requests to the Hub -even methods that don't necessarily require authentication- will use your
-access token by default. If you want to disable implicit use of your token, you should set the
-`HF_HUB_DISABLE_IMPLICIT_TOKEN` environment variable.
+Please be careful when passing tokens as parameter. It is always best practice to load the token from a secure vault instead of hardcoding it in your codebase or notebook. Hardcoded tokens present a major risk of leak if you share your code inadvertently.
```

Comment:
Yes! We'll probably need to do a few updates in the libraries that depend on it as we were encouraging this for now